js/classes/entities/midground/weapon/projectile.js

```diff
@@ -11,9 +11,14 @@ class Projectile {
      * 
      * @param {number} type the Projectile type (please use Projectile.WOOD, Projectile.BOMB, etc.)
      * @param {Vector} pos the position at which the projectile should begin
-     * @param {Vector} target the target position
+     * @param {Vector} [target] (OPTIONAL) the target position
+     * @throws {Error} Will throw an error if type is not a Projectile member type.
      */
     constructor(type, pos, target) {
+        if (typeof type !== "number" || type % 1 !== 0 || type < 0 || type > 5) {
```

Comment:
Nice job error checking however I am a bit lost on why the type cannot equal "number" and how that means what it says it means. I am also lost on what the necessity of type % 1 !== 0 is or what that does. Could you explain?

Reply:
Since the if statement is throwing an error for an invalid type, the condition is checking for an invalid type (not a `number`). `type % 1 !== 0` is not intuitive to me either but since a `number` in JS can be either decimal or an integer, `type % 1 !== 1` checks if the value in an integer or not.